Code/Framework/AzCore/AzCore/Math/Sfmt.cpp

```diff
@@ -392,6 +392,8 @@ namespace AZ
     //=========================================================================
     void Sfmt::Seed(AZ::u32* keys, int numKeys)
     {
+        AZStd::lock_guard<decltype(m_generationMutex)> lock(m_generationMutex);
```

Comment:
Can you update this function to accept a `const AZ::u32*`?
The keys aren't being modified by this function.
```suggestion
    void Sfmt::Seed(const AZ::u32* keys, int numKeys)
    {
        AZStd::scoped_lock lock(m_generationMutex);
```

Reply:
Also fixed the constructor.

Reply:
Thanks!